src/darwin/Framework/CHIP/MTRBaseDevice.h

```diff
@@ -130,6 +130,37 @@ typedef NS_ENUM(uint8_t, MTRTransportType) {
     MTRTransportTypeTCP,
 } API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4));
 
+/**
+ * A path indicating a attribute on a device.
+ * If value is a wildcard, set it to nil.
+ */
+MTR_NEWLY_AVAILABLE;
+@interface MTRAttributeRequestPath : NSObject <NSCopying>
+@property (nonatomic, readonly, copy) NSNumber * _Nullable endpoint;
+@property (nonatomic, readonly, copy) NSNumber * _Nullable cluster;
+@property (nonatomic, readonly, copy) NSNumber * _Nullable attribute;
+
++ (MTRAttributeRequestPath *)requestPathWithEndpointID:(NSNumber * _Nullable)endpointID
+                                             clusterID:(NSNumber * _Nullable)clusterID
+                                           attributeID:(NSNumber * _Nullable)attributeID MTR_NEWLY_AVAILABLE;
+@end
+
+/**
+ * A path indicating a event on a device.
+ * If value is a wildcard, set it to nil.
```

Comment:
* A path indicating an event being requested (for read or subscribe).  
 *
 * nil is used to represent wildcards.

Reply:
modified